File: Week_01/kotlin/src/_925_LongPressedName.kt
class _925_LongPressedName_ {
    class Solution {
        fun isLongPressedName(name: String, typed: String): Boolean {
            val m = name.length
            val n = typed.length
            var i = 0
            var j = 0
            while (j < n) {
                when {
                    i < m && name[i] == typed[j] -> {
                        i++
                        j++
                    }
                    j > 0 && typed[j] == typed[j - 1] -> {
                        j++
                    }
                    else -> {
                        return false
                    }
                }
            }
            return i == m
        }
    }
}

File: Week_01/js/src/_925_LongPressedName.js
/**
 * @param {string} name
 * @param {string} typed
 * @return {boolean}
 */
var isLongPressedName = function(name, typed) {
    const m = name.length, n = typed.length
    let i = 0, j = 0
    while (j < n) {
        if (i < m && (name[i] === typed[j])) {
            i++
            j++
        } else if (j > 0 && typed[j] === typed[j - 1]) {
            j++
        } else {
            return false
        }
    }
    return i == m
};